client/data/help/use-help-search-query.ts

```diff
@@ -14,41 +14,77 @@ export interface SearchResult {
 	source?: string;
 }
 
+interface TailoredArticles {
+	post_ids: Array< number >;
+	blog_id: number;
+	locale: string;
+}
+
 interface APIFetchOptions {
 	global: boolean;
 	path: string;
 }
 
-const fetchArticlesAPI = ( search: string, locale: string, sectionName: string ) => {
-	const queryString = buildQueryString( {
-		query: search,
-		locale,
-		section: sectionName,
+const fetchArticlesFromWPCom = async ( queryString: string ): Promise< SearchResult[] > => {
+	return await wpcomRequest( {
+		path: `help/search/wpcom?${ queryString }`,
+		apiNamespace: 'wpcom/v2/',
+		apiVersion: '2',
 	} );
+};
 
-	if ( canAccessWpcomApis() ) {
-		return wpcomRequest( {
-			path: `help/search/wpcom?${ queryString }`,
-			apiNamespace: 'wpcom/v2/',
-			apiVersion: '2',
-		} );
-	}
-
-	return apiFetch( {
+const fetchArticlesFromAPI = async ( queryString: string ): Promise< SearchResult[] > => {
+	return await apiFetch( {
 		global: true,
 		path: `/help-center/search?${ queryString }`,
 	} as APIFetchOptions );
 };
 
+const fetchArticlesFromWP = async ( articles: TailoredArticles ): Promise< SearchResult[] > => {
+	const { post_ids, blog_id, locale } = articles;
+	return await wpcomRequest( {
+		path: `help/articles`,
+		apiNamespace: 'wpcom/v2/',
+		apiVersion: '2',
+		method: 'PUT',
+		body: { post_ids, blog_id, locale },
+	} );
+};
+
+const fetchArticlesAPI = async (
+	search: string,
+	locale: string,
+	sectionName: string,
+	articles: TailoredArticles | undefined
+): Promise< SearchResult[] > => {
+	const queryString = buildQueryString( { query: search, locale, section: sectionName } );
+	let articlesResponse: SearchResult[] = [];
+	let searchResultResponse: SearchResult[] = [];
+
+	if ( articles ) {
+		articlesResponse = await fetchArticlesFromWP( articles );
+	}
+
+	if ( canAccessWpcomApis() ) {
+		searchResultResponse = await fetchArticlesFromWPCom( queryString );
+	} else {
+		searchResultResponse = await fetchArticlesFromAPI( queryString );
+	}
```

Comment:
I think some renaming is necessary, as it's a little confusing, for example it is not clear the difference between `fetchArticlesFromWP` and `fetchArticlesFromWPCom`!

`fetchArticlesFromWPcom` and `fetchArticlesFromAPI` perform a search for a specific query. The quickest way to solve this would be to have the two calls in this if-else.
If you really want to have them separate, we need to rename them:
- fetchArticlesFromWPcom => searchQueryWpcom
- fetchArticlesFromAPI => searchQueryAPI

These are just suggestions, feel free to propose something else.

Regarding `fetchArticlesFromWP`, what if we cannot access Wpcom APIs? We need the same thing as for the other calls in this function :) And that's why we need an ETK API for this, like we did in 
`searchResultResponse = await fetchArticlesFromAPI( queryString );`